app/src/main/java/fr/free/nrw/commons/settings/Prefs.java

```diff
@@ -9,6 +9,7 @@
     public static final String IS_CONTRIBUTION_COUNT_CHANGED = "ccontributionCountChanged";
     public static final String MANAGED_EXIF_TAGS = "managed_exif_tags";
     public static final String KEY_LANGUAGE_VALUE = "languageDescription";
+    public static final String KEY_THEME_VALUE = "theme";
```

Comment:
You are using a different key than the one used `preferences.xml`, is this intentional?

Reply:
@macgills Yes! It is intentional

Reply:
It is however the same as the pre-existing boolean key... I am actually unsure what happens when you migrate the type of a sharedPreference value. Have you installed a previous version and then upgraded to this one after having changed the night mode value? I wouldn't be surprised if something explodes.

Also please elaborate on why it is intentional.

Reply:
@macgills I need to change it because I was getting the error String can't be converted into Boolean. The app data needs to be cleaned once after the update.

Reply:
Okay, the code will still break and we do not ask our users to clean their data to get an update for our app.

Let me explain a bit how PreferenceFragment works.
It loads a preference from xml eg
```
        <fr.free.nrw.commons.ui.LongTitlePreferences.LongTitleSwitchPreference
            android:key="displayNearbyCardView"
            android:title="@string/display_nearby_notification"
            android:defaultValue="true"
            android:summary="@string/display_nearby_notification_summary" />
````
PreferenceFragment by default uses the `DefaultSharedPreferences` to store and retrieve these values, for the above example `sharedPrefs.getBoolean("displayNearbyCardView", "true")` would be used when the UI loads.
It also uses the sharedPreferences to save this value when the UI updates `sharedPrefs.putBoolean("displayNearbyCardView", <uiValue>)`. This code is internal to PreferenceFragment.

The original `key` of this preference was `theme` when it was a boolean, you try to read it in LoginActivity as a String with the same key. If this preference exists ie set from a previous install, I forsee this being an issue.

I recommend you use an entirely new key, and set `android:entries` `android:entryValues` `android:defaultValue"` in xml

I don't particularly like PreferenceFragment but it is here so we should use it. 

Augh we are using the deprecated PreferenceFragment instead of Androidx even! You don't have to worry about that though

Reply:
Please review the new changes and let me know if I understood correctly.